src/taylor.hpp

```diff
@@ -469,15 +469,20 @@ inline void taylor<5, simd_vector>::set_basis(const std::array<simd_vector, NDIM
 
     const T r2 = sqr(X[0]) + sqr(X[1]) + sqr(X[2]);
     T r2inv = 0.0;
-// #if !defined(HPX_HAVE_DATAPAR)
+#if !defined(HPX_HAVE_DATAPAR)
     for (volatile integer i = 0; i != simd_len; ++i) {
         if (r2[i] > 0.0) {
             r2inv[i] = ONE / std::max(r2[i], 1.0e-20);
         }
     }
-// #else
-//     where(r2 > 0.0) | r2inv = ONE / r2;
-// #endif
+#else
+    #ifdef Vc_IS_VERSION_2  
+        where(r2 > 0.0, r2inv) = ONE / r2;
+        using Vc::sqrt;
+    #else
+        where(r2 > 0.0) | r2inv = ONE / r2;
```

Comment:
We found out the hard way, that `where` works on assignment only. The intention was to perform only the _calculation_ for the elements which are non-negative, but as it turns out the code will perform the division for all elements, masking the assignment only.